artman/utils/protoc_utils.py

```diff
@@ -198,21 +198,36 @@ def proto_compiler_command(self):
 }
 
 
-def group_by_dirname(protos):
-    """Groups the file paths by direct parent directory.
+def group_by_go_package(proto_files):
+    """Groups the file paths by `option go_package` in the file.
+    This reflects the logic in https://github.com/google/go-genproto/blob/master/regen.go
 
     Returns:
-        A dict mapping from the directory name to the list of proto files in
-        it.
+        A dict mapping go_package to the list of proto files in the package.
     """
-    dirs = {}
-    for proto in protos:
-        dirname = os.path.dirname(proto)
-        if dirname not in dirs:
-            dirs[dirname] = [proto]
-        else:
-            dirs[dirname].append(proto)
-    return dirs
+
+    def go_pkg(file):
+        prefix = 'option go_package ='
+        with open(file, encoding='utf-8') as f:
```

Comment:
Is this line the only line that's changed from the first PR?

Reply:
Yup!